### PR TITLE
Dockerfile: Install libyang

### PR DESCRIPTION
**Pull request by cfra**
Recent versions of FRR require libyang to build. So install it from
the FRR CI servers.

Signed-off-by: Christian Franke <chris@opensourcerouting.org>

### CODE REVIEW

**Comment by rzalamena** [comment]
I see two options here:

1. Wait for `libyang` matter to settle down (e.g. start seeing distro packages);
2. Move the docker package somewhere fixed (unless we make sure that URL you've used is safe);

If we want the docker part merged as soon as possible, the option (2) seems reasonable.

**Comment by cfra** [comment]
I have asked whether this URL is safe in #mainters and have not got gotten a reply yet.
As it contains a build number, I think it should be okay, but I guess @mwinter-osr would know best.

Any comments?

**Comment by mwinter-osr** [comment]
CI:rerun

File: Dockerfile [hunk 1]
```diff
@@ -30,13 +31,24 @@ RUN export DEBIAN_FRONTEND=noninteractive \
...
+        wget \
         x11-xserver-utils \
         xterm \
     && pip install \
         exabgp==3.4.17 \
         ipaddr \
         pytest
 
+RUN cd /tmp \
+    && wget -q https://ci1.netdef.org/artifact/LIBYANG-YANGRELEASE/shared/build-1/Ubuntu-18.04-x86_64-Packages/libyang-dev_0.16.46_amd64.deb \
```
Comment (mwinter-osr) [inline Dockerfile]: Better would be
`https://ci1.netdef.org/browse/LIBYANG-YANGRELEASE/latestSuccessful/artifact/shared/Ubuntu-16.04-x86_64-Packages/libyang-dev_0.16.46_amd64.deb`

(`build-1` is result from the first build on the testplan. Using the above redirects to last successful run of the testplan)

File: Dockerfile [hunk 2]
```diff
@@ -30,13 +31,24 @@ RUN export DEBIAN_FRONTEND=noninteractive \
...
+         -O libyang-dev.deb \
+    && wget -q https://ci1.netdef.org/artifact/LIBYANG-YANGRELEASE/shared/build-1/Ubuntu-18.04-x86_64-Packages/libyang_0.16.46_amd64.deb \
```
Comment (mwinter-osr) [inline Dockerfile]: Same here.
Use `https://ci1.netdef.org/browse/LIBYANG-YANGRELEASE/latestSuccessful/artifact/shared/Ubuntu-16.04-x86_64-Packages/libyang_0.16.46_amd64.deb`

File: Dockerfile (hunk 1 above)
Comment (cfra) [reply Dockerfile]: As discussed with Martin, as this path also contains the package version, let's keep it at the old url, until we have a PPA.

File: Dockerfile (hunk 2 above)
Comment (cfra) [reply Dockerfile]: See previous comment.

**Comment by cfra** [comment]
This actually still needs some changes. For some reason the `frr_user_types` plugin doesn't get built or installed correctly, yet.

**Comment by cfra** [comment]
Build with libyang is now working correctly, so I think this could be merged.

**Pull request merged**